core/domain/story_services.py

```diff
@@ -970,19 +970,32 @@ def record_completed_node_in_story_context(
         progress_model.put()
 
 
-def get_chapter_notification_dicts() -> Dict[str, List[Dict[
-    str, Sequence[str]]]]:
-    """Returns a dict of behind schedule and upcoming chapters."""
+def get_chapter_notification_dicts() -> Tuple[
+    List[story_domain.OverdueStoryDict],
+    List[story_domain.UpcomingStoryDict]]:
+    """Returns a dict of behind-schedule and upcoming chapters.
+
+    Returns:
+        tuple(list(OverdueStoryDict), list(UpcomingStoryDict)). A 2-tuple
+        whose first element is a list of story dicts having overdue chapters
+        and second element is a list of story dicts having upcoming chapters.
+    """
     topic_models = topic_fetchers.get_all_topics()
-    overdue_stories_dicts = []
-    upcoming_stories_dicts = []
+    overdue_stories_dicts: List[story_domain.OverdueStoryDict] = []
+    upcoming_stories_dicts: List[story_domain.UpcomingStoryDict] = []
+    total_canonical_story_ids = []
```

Comment:
all_canonical_story_ids (only use "total" for counts).

Reply:
Done